libweston/backend-rdp/rdprail.c

```diff
@@ -4235,42 +4610,35 @@ rdp_rail_notify_app_list(void *rdp_backend, struct weston_rdprail_app_list_data
 				goto Exit_updatePath;
 			iconData.iconFormat = RDPAPPLIST_ICON_FORMAT_BMP;
 			iconData.iconBitsLength = iconData.iconHeight * iconData.iconStride;
-			iconData.iconBits = malloc(iconData.iconBitsLength);
-			if (!iconData.iconBits)
-				goto Exit_updatePath;
-			char *src = (char *)pixman_image_get_data(app_list_data->appIcon);
+			iconData.iconBits = xmalloc(iconData.iconBitsLength);
+			char *src = pixman_image_get_data(app_list_data->appIcon);

```

Comment:
pixman_image_get_data returns uint32_t *, so don't we need cast to avoid warning?

Reply:
Really wishing I hadn't touched any of the applist code. :)
Correct, again.